testhelper/tests.go

```diff
@@ -58,8 +58,14 @@ func (options *TestOptions) checkConsistency(plan *terraform.PlanStruct) {
 			changesJson = string(changesBytes)
 		}
 
-		resourceDetails := fmt.Sprintf("Name: %s Address: %s Actions: %s\nDIFF:\n%s\n\nChange Detail:\n%s", resource.Name, resource.Address, resource.Change.Actions, GetBeforeAfterDiff(changesJson), changesJson)
+		var resourceDetails string
 
+		if resource.Change.Actions.Update() {
+			resourceDetails = fmt.Sprintf("Name: %s Address: %s Actions: %s\nDIFF:\n%s\n\nChange Detail:\n%s", resource.Name, resource.Address, resource.Change.Actions, GetBeforeAfterDiff(changesJson), changesJson)
+		} else {
+			// Do not print changesJson because might expose secrets
```

Comment:
well not true - we don't need to do a diff for any other type of action, only update

Reply:
So there are two outputs here the diff and the changes json. The changes json may be useful for debugging issues with creating or deleting. But it would still have the potential to leak secrets.